### PR TITLE
Added support for mpv if mplayer is not available.

### PR DESCRIPTION
**Pull request by mrkrossxdx**
This commit simply try to use mpv if mplayer is not found.

### CODE REVIEW

File: youtube_dl/downloader/mplayer.py [hunk 1]
```diff
@@ -16,11 +16,18 @@ def real_download(self, filename, info_dict):
...
+            args = [
+                'mplayer', '-really-quiet', '-vo', 'null', '-vc', 'dummy',
+                '-dumpstream', '-dumpfile', tmpfilename, url]
+                        
+        # Check for mpv
+        elif check_executable('mpv', ['-h']):
+            args = [
+                'mpv', '-really-quiet', '--vo=null', '--stream-dump=' + tmpfilename, url]
+        else:
             self.report_error('MMS or RTSP download detected but "%s" could not be run' % args[0])
```
Comment (dstftw) [inline youtube_dl/downloader/mplayer.py]: This will fail if neither mplayer nor mpv is available.

**Comment by mrkrossxdx** [comment]
I'm sorry, but I don't understand what you mean, is the original code supposed to work also if mplayer is not available?

I tried to download a MMS stream and youtube-dl fails complaining about the absence of mplayer. I have writes a small shell script that calls mpv with the correct parameters and everything worked fine, so I had assumed that mplayer is required for download MMS stream, but given my poor knowledge of the codebase I can be wrong.

Thank you for taking the time to review this.

Best regards,

Marco

**Comment by dstftw** [comment]
[`args` is empty list](https://github.com/rg3/youtube-dl/pull/5513/files#diff-ce2afa610d98a8227657cea63af36563R19), [access to 0 item of empty list](https://github.com/rg3/youtube-dl/pull/5513#discussion-diff-29000180R31) => `IndexError`

**Comment by mrkrossxdx** [comment]
D'oh, I don't have noted that the error message use the args list, thank you very much for this clarification. I will create another pull request with the correct error message.

**Pull request closed without merging**